### ISSUE TITLE
Consider dry-run the release pipeline for each PR

### ISSUE THREAD

**Issue by Mossaka** [spinkube/containerd-shim-spin#206]
any change to the release pipeline could break it without getting noticed.

**Comment by kate-goldenring** [comment]
Another option is to ensure folks test out their changes to the release pipeline in a fork and link a successful run in order to merge the PR

**Comment by Mossaka** [comment]
yeah, we need to ask reviewer / maintainer of this repo to especially pay attention to any changes to the `release.yaml`
